lib/actions-oclif/envs.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright 2016-2019 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Command, flags } from '@oclif/command';
+import { stripIndent } from 'common-tags';
+import * as _ from 'lodash';
+
+import * as cf from '../utils/common-flags';
+import { CommandHelp } from '../utils/oclif-utils';
+
+type ApplicationVariable = import('balena-sdk').ApplicationVariable;
+type DeviceVariable = import('balena-sdk').DeviceVariable;
```

Comment:
Why do you need to alias these types? Doesn't the following work?

```
import { ApplicationVariable, DeviceVariable } from 'balena-sdk';
```

Reply:
I hadn't thought about it in terms of "aliasing" the types, and yes what you've suggested works, but... I meant to exercise a TypeScript 2.9 feature of importing types only, without generating a `require` statement in transpiled JS, while avoiding the "need" for two `import 'balena-sdk'` statements: one at the top of the file for types only, and one inside the `run()` function for the runtime importing of the module.

I understand that the import statement you suggested is also "likely" to avoid generating a `require` statement in transpiled JS (depending on the `tsc` being sufficiently clever), but it leads to CLI code as follows:
```
import { ApplicationVariable, DeviceVariable } from 'balena-sdk';

export default class EnvsCmd extends Command {
	public async run() {
		const balena = (await import('balena-sdk')).fromSharedOptions();
```

One would look at the code above and think (like I myself did in the past) _"oh why have that inner `await import('balena-sdk')` if there's already an `import` statement at the top of the file? Let me "fix" this "code duplication" by moving everything to the top of the file:"_
```
import { ApplicationVariable, DeviceVariable, fromSharedOptions } from 'balena-sdk';

export default class EnvsCmd extends Command {
	public async run() {
		const balena = fromSharedOptions();
```
But the latter would actually cause a `require` statement to be generated at the top of the file in transpiled code. (And we generally want to avoid that in order to reduce CLI start time, even though oclif's on-demand importing of command files makes it less of an issue than with Capitano.)

Anyway, in summary, the reason for using `type x = import('y').x` is to make it clear (?) and explicit that it's a type import only, without producing a `require` in transpiled code. Of course, I wasn't that successful if you had to ask the question... :-)

Reply:
@pdcastro if you need to convey meaning, you can do so with a comment. The format @dfunckt suggested makes a lot more sense and is the pattern we use throughout every other project in the company.

Reply:
Code changed as suggested, thanks.